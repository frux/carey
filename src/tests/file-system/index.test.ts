import { assertMigration } from '../helpers';

describe('Actions', () => {
	describe('move', () => {
		test('should rename file', () => assertMigration(
			{
				foo: {
					'bar.txt': 'test',
				},
			},
			({ move }) => ([
				{
					with: '/foo/bar.txt',
					do: [
						move('/foo/baz.txt'),
					],
				},
			]),
		));

		// Skipped because of https://github.com/streamich/memfs/issues/397
		test.skip('should move file', () => assertMigration(
			{
				foo: {
					'bar.txt': 'test',
				},
			},
			({ move }) => ([
				{
					with: '/foo/bar.txt',
					do: [
						move('/bar.tsx'),
					],
				},
			]),
		));
	});
});
