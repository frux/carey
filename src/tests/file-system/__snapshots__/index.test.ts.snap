// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions move should move file 1`] = `
Object {
  "/bar.tsx": "test",
}
`;

exports[`Actions move should rename file 1`] = `
Object {
  "/foo/baz.txt": "test",
}
`;
